### ISSUE TITLE
[new-model] utility-model family (`utility()`) built on the m3 backend

### ISSUE THREAD

**Issue by GidonFrischkorn** [venpopov/bmm#358]
Following up on our discussion about utility models in bmm, this issue is the concrete proposal. The aim is a first attempt at integrating utility-theoretic choice models (random-utility, expected utility, probability weighting, welfare weights for economic games) as a model family, and at finding out what API serves these models best. Nothing here is final, the surface is explicitly up for discussion.

The starting observation: the m3 backend already does most of the work. The softmax rule is formally a random-utility (logit) model, and `choice_rule = "simple"` generates `log(activation)` activations, that is the Luce ratio rule over raw utilities, so utility models with fixed payoff structures run natively today. What is missing is API surface: per-category formula generation (a shared-slope formula runs but answers the wrong question), default priors for utility parameters, and design-dependent identifiability checks that currently fail silently (flat posteriors).

A concrete use case that motivated this: welfare-weight models for nested social dilemmas (Gross et al., 2025, *Communications Psychology*, https://doi.org/10.1038/s44271-025-00192-y), where choice options have known payoff vectors and the parameters of interest are ingroup/outgroup welfare weights with directly interpretable thresholds. The original paper used logistic regressions; a utility-parametrized multinomial model is a genuine extension.

## What the exploration established

I ran a seven-round exploration on my fork (all code, fits, and the design memo are merged there: [local/exploration/m3-utility](https://github.com/GidonFrischkorn/bmm/tree/develop/local/exploration/m3-utility), summary in `DESIGN_utility_api.md`). The short version:

- **Identifiability is measured, not assumed.** Power utility (gamma, rho) recovers jointly given sufficient value range (range ratio >= 3 as a provisional criterion); Prelec alpha recovers with variable set sizes and shows no false positives at the alpha = 1 null; the composite value-plus-weighting model recovers all parameters at `adapt_delta = 0.95` despite a strong posterior correlation between c and alpha (0.906). Each condition has a corresponding guard (G1–G4) prototyped in `16_identifiability_guards.R`, with measured justification rather than analytic hand-waving.
- **The welfare-weight case runs through the existing machinery end-to-end**: `m3(choice_rule = "simple")` + identity links + `fixed_parameters$b` as numeraire recovers the weights cleanly (`22_welfare_bmm_simple.R`). No core changes were needed — which is exactly why I think this should be API, not new backend.

## Proposed API (first attempt — please push back)

A first-class constructor `utility()` in `R/model_utility.R`, with the m3 machinery inherited via class stacking — `c("bmmodel", "m3", "utility", "utility_<version>")` — so `bmm()`, the choice rules, and `fixed_parameters` handling come for free, per the documented adding-new-models path:

```r
# Welfare weights from a known payoff matrix (economic games)
model <- utility(
  resp_cats   = c("nkeep", "ningroup", "nuniversal"),
  num_options = c(1, 1, 1),
  payoffs     = rbind(nkeep      = c(b = 1.0, wi = 0.0, wo = 0.0),
                      ningroup   = c(b = 0.5, wi = 1.0, wo = 0.0),
                      nuniversal = c(b = 0.3, wi = 0.6, wo = 0.9)),
  choice_rule = "simple",
  numeraire   = c(b = 1)          # fixes w_self = 1 as the unit of account
)

# Value-directed designs (per-trial value columns, EU / prospect-theory pieces)
model <- utility(
  resp_cats   = c("corr", "other", "npl"),
  num_options = c(1, 1, 8),
  value_cols  = c(corr = "V"),
  utility     = "power",          # adds rho
  weighting   = "prelec"          # adds alpha
)
```

The constructor generates the per-category activation formulas (so the shared-slope trap cannot happen), sets links and default priors, and `check_data.utility` runs the identifiability guards automatically at fit time — silent non-identifiability becomes a loud, early message.

On naming: I would export `utility()` rather than `m3_utility()`. The audience for these models (utility theory, behavioral economics) does not know the M3, and the user surface should speak random-utility vocabulary, with the m3 backend as an implementation detail. However, I see the counterargument — `utility` is a generic word to claim in the namespace — so if you prefer `m3_utility()` or another name, the change is a one-line class/alias matter and I am happy either way.

## Scope

Stage 1 covers category-count choice (the designs above). Per-option attribute choice — binary/lottery prospect theory à la Nilsson et al. (2011) — is deliberately out of scope: the m3 backend cannot express per-option attributes, and bolting it on now would distort the API. The constructor signature is designed so that case can land later (e.g., a `version = "lottery"` with its own data interface) without breaking changes.

## Proposed process

I have a full implementation spec with tests, recovery validation, and a vignette outline on my fork ([GidonFrischkorn/bmm#18](https://github.com/GidonFrischkorn/bmm/issues/18)). My suggestion: I implement and validate it on the fork first and bring it here as a PR, with this issue as the place to settle the API questions — naming, the `payoffs`/`value_cols` split, guard severities, and anything else you would do differently. Treat all of it as a first attempt at finding the right surface for these models.
